app/assets/src/components/views/compare/SamplesHeatmapVis.jsx

```diff
@@ -116,6 +124,28 @@ class SamplesHeatmapVis extends React.Component {
     });
   }
 
+  generateHeatmapCaptions = () => {
+    const { thresholdFilters } = this.props;
+
+    const numFilters = size(thresholdFilters);
+
+    if (numFilters == 0) {
+      return [];
+    }
+
+    const filterStrings = map(
+      filter => `${filter.metricDisplay} ${filter.operator} ${filter.value}`,
+      thresholdFilters
+    );
+
+    const fullString = `${numFilters} filter${
+      numFilters > 1 ? "s were" : " was"
+    } applied to the above heatmap: ${filterStrings.join(", ")}.
+      Non-conforming cells have been hidden or grayed out.`;
+
+    return splitIntoMultipleLines(fullString, CAPTION_LINE_WIDTH);
```

Comment:
Any idea if we could set the max number of characters having the font-size (could be exported) and width of heatmap (can be retrieved from the svg) into account? Would be cool to adapt to wider and shorter heatmaps.

Reply:
In any case, make sure current width works well with less or more samples.

Reply:
Since we're not using a monospace font and characters can have varying widths, it's hard to take the font-size and width of the heatmap into account.

The heatmap has min-width 1240 (even if there are fewer samples, the sample columns will get wider), so the caption line width of 100 seems to work well for that.